### ISSUE TITLE
IEnumerable fails to serialize with the Json serializer

### ISSUE THREAD

**Issue by mastoj** [Particular/NServiceBus#2028]
We had a message in 4.0.4 that used IEnumerable<T> as a property. This worked just fine in 4.0.4, but in 4.4.2 it stopped working. The problem seems to be that it is not possible to use Enumerables derived from a linq expression as properties in a message any longer. Is that how it should be?

I tried both: 

    Bus.Send(new StartSaga() { Text = line, EnumerableOfThings = new List<string>{"abc", "abb", "bcd", "add", "ccc"}.Select(y => y)});

and 

    Bus.Send(new StartSaga() { Text = line, EnumerableOfThings = new List<string>{"abc", "abb", "bcd", "add", "ccc"}.Where(y => y.StartsWith("a"))});

and they work in 4.0.4 but I get a serialization error in 4.4.2.

**Comment by johnsimons** [comment]
Have just tried against json.net v6.0.8, still broken.
Do I rollback json.net to v4.5.11, the version that used to work on ?
@Particular/nservicebus thoughts?

**Comment by SimonCropp** [comment]
@johnsimons changing json.net backwards can only be done in a major release since technically we could be removing features

if they use this https://github.com/SimonCropp/NServiceBus.Newtonsoft.Json they can pick their json.net version. so possible workaround?

**Comment by synhershko** [comment]
@johnsimons there's no need to go backwards, the fix is rather simple, see https://github.com/Particular/NServiceBus/issues/2028#issuecomment-40098434. I can take send a PR if you want.

**Comment by johnsimons** [comment]
@synhershko according to your comment the fix is for the user to initialize the prop with a type that has a ctor, so not sure what PR you can send ?

**Comment by johnsimons** [comment]
@SimonCropp we can downgrade json lib in the v6 release ?

**Comment by SimonCropp** [comment]
@johnsimons yes but u would need to document all the things fixed and added between the two json .net versions as breaking changes. since in effect they are breaking changes from the perspective of people consuming NSB

**Comment by synhershko** [comment]
@johnsimons see my last comment there, namely http://stackoverflow.com/a/8372749/135701. We can have this config enabled by default - but this _may_ be a breaking change to some who rely on it so maybe require enabling it explicitly?

**Comment by johnsimons** [comment]
@SimonCropp yes, so do we go ahead and do that?

**Comment by SimonCropp** [comment]
@johnsimons i think @andreasohlund has plans to dump Json.net from the core. perhaps wait for him to weigh in??

**Comment by johnsimons** [comment]
@synhershko i guess the fact is that this used to work and no longer works now

**Comment by SimonCropp** [comment]
@johnsimons https://github.com/Particular/NServiceBus/issues/2460

**Comment by johnsimons** [comment]
@SimonCropp this issue has previously been closed because of that, and then reopened again see https://github.com/Particular/NServiceBus/issues/2028#issuecomment-66434623, so are we saying we should not fix this now?

**Comment by SimonCropp** [comment]
@johnsimons so the fix suggested by @synhershko works and wont break anything else (to the best of our knowledge)? if so then yes it should go in a patch

**Comment by johnsimons** [comment]
ok, I must be blind, but what is the fix suggested by @synhershko ?
Is it for the user to use `List` ?

**Comment by synhershko** [comment]
@johnsimons something like this: http://stackoverflow.com/a/17680641/135701

**Comment by johnsimons** [comment]
I think the best way forward is a PR to see if we can fix this without any user changes

**Comment by synhershko** [comment]
@johnsimons on it, will send something next week

**Comment by SimonCropp** [comment]
@synhershko bump

**Comment by andreasohlund** [comment]
We've decided to not fix this due to the risk of introducing further regressions.

This situation is due to the historical decision to ilmerge json.net in the core. Version 6 of the core will handle this by providing an external json.net serializer that puts the user in control of the exact version of json.net, or alternatively, change the behavior using the standard json.net api's.

An example on how this can be done can be seen [here](https://github.com/SimonCropp/NServiceBus.Newtonsoft.Json)

Closing this one as `Won't fix`
